vespalib/src/vespa/vespalib/fuzzy/fuzzy_matcher.cpp

```diff
@@ -6,8 +6,34 @@
 #include <vespa/vespalib/text/lowercase.h>
 #include <vespa/vespalib/text/utf8.h>
 
-vespalib::FuzzyMatcher vespalib::FuzzyMatcher::from_term(std::string_view term, uint32_t maxEditDistance, uint32_t prefixLength) {
-    return FuzzyMatcher(LowerCase::convert_to_ucs4(term), maxEditDistance, prefixLength);
+namespace {
+
+    std::vector<uint32_t> cased_convert_to_ucs4(std::string_view input) {
+        std::vector<uint32_t> result;
+        result.reserve(input.size());
+        vespalib::Utf8Reader reader(input.data());
+        while (reader.hasMore()) {
+            result.emplace_back(reader.getChar());
+        }
+        return result;
+    }
+
+} // anonymous
+
+vespalib::FuzzyMatcher::FuzzyMatcher():
+        _max_edit_distance(DefaultMaxEditDistance),
+        _prefix_size(DefaultPrefixSize),
+        _is_cased(false)
+{}
+
+vespalib::FuzzyMatcher::FuzzyMatcher(std::string_view term, uint32_t max_edit_distance, uint32_t prefix_size, bool is_cased):
+        _max_edit_distance(max_edit_distance),
+        _prefix_size(prefix_size),
+        _is_cased(is_cased)
+{
+    _folded_term_codepoints = _is_cased ? cased_convert_to_ucs4(term) : LowerCase::convert_to_ucs4(term);
```

Comment:
Please move initialization of these member variables to the ctor initializer lists.